### PR TITLE
Refine common code snippet of samples

### PR DESCRIPTION
**Pull request by D-W-**
# PR into Azure/azureml-examples

Add cpu-cluster creation code snippet
Support both ml_client config file & user input subid,workspace, rg.
add more comments to the MLClient init part to present all login options to customer, and explain when use which

## Checklist

I have:

- [ ] read and followed the contributing guidelines

## Changes

-

fixes #

### CODE REVIEW

File: sdk/jobs/pipelines/1a_pipeline_with_components_from_yaml/pipeline_with_components_from_yaml.ipynb [hunk 1]
```diff
@@ -61,10 +95,42 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "#Enter details of your AML workspace\n",
-    "subscription_id = '<SUBSCRIPTION_ID>'\n",
-    "resource_group = '<RESOURCE_GROUP>'\n",
-    "workspace = '<AML_WORKSPACE_NAME>'"
+    "config_path = \"../../../../.azureml/config.json\"\n",
```
Comment (wangchao1230) [inline sdk/jobs/pipelines/1a_pipeline_with_components_from_yaml/pipeline_with_components_from_yaml.ipynb:98]: what's this path?

File: sdk/jobs/pipelines/1a_pipeline_with_components_from_yaml/pipeline_with_components_from_yaml.ipynb [hunk 2]
```diff
@@ -50,7 +49,42 @@
...
+    "try:\n",
+    "    credential = DefaultAzureCredential(exclude_visual_studio_code_credential=True)\n",
+    "    # Check if given credential can get token successfully.\n",
+    "    credential.get_token('https://management.azure.com/.default')\n",
+    "except Exception as ex:\n",
+    "    # If exception happens when retrieve token, try exclude the failed credential like this then try again:\n",
+    "    # Exclude VSCode credential:\n",
+    "    # credential = DefaultAzureCredential(exclude_visual_studio_code_credential=True)\n",
+    "    # Exclude \n",
+    "    # credential = DefaultAzureCredential(exclude_environment_credential=True)\n",
```
Comment (wangchao1230) [inline sdk/jobs/pipelines/1a_pipeline_with_components_from_yaml/pipeline_with_components_from_yaml.ipynb:78]: when will environment credential be set?

**Comment by wangchao1230** [comment]
Add example change of CI workflow yml files

File: sdk/jobs/pipelines/1a_pipeline_with_components_from_yaml/pipeline_with_components_from_yaml.ipynb (hunk 1 above)
Comment (D-W-) [reply sdk/jobs/pipelines/1a_pipeline_with_components_from_yaml/pipeline_with_components_from_yaml.ipynb:98]: Root level of this project, maybe add a comment?

File: sdk/jobs/pipelines/1a_pipeline_with_components_from_yaml/pipeline_with_components_from_yaml.ipynb (hunk 2 above)
Comment (D-W-) [reply sdk/jobs/pipelines/1a_pipeline_with_components_from_yaml/pipeline_with_components_from_yaml.ipynb:78]: It works when some specific environment variables are set, eg AZURE_TENANT_ID, AZURE_CLIENT_ID, etc. ref: https://docs.microsoft.com/en-us/dotnet/api/azure.identity.environmentcredential?view=azure-dotnet

**Pull request merged**